### PR TITLE
Deprecation clean-up

### LINKED ISSUE #762: undo deprecation of `step_kpca()`
it was deprecated in version 0.1.7

### PR DESCRIPTION
**Pull request by hfrick**
This PR does a few things with regard to deprecation and sneaks in a bonus clean-up of the pkgdown site:

* undoes the deprecation of `step_kpca()`; closes #762
* escalates the deprecation of the `preserve` argument to `step_pls()` and `step_dummy()`
* adds lifecycle-style deprecation for the `options` argument to `step_nzv()` so it won't get overlooked next time 
* tidies up the pkgdown site so that the newly exported `recipes_eval_select()` shows up in the section for internal tools rather than at the top and removes the quadruplication of the section on selection via role

The imputation steps and the steps for up/downsampling were already in the correct stage of the deprecation process.

### CODE REVIEW

File: R/nzv.R
Comment (hfrick) [inline R/nzv.R:89]: this has been deprecated for quite a while but since it only ever sent a message, but no warning or error, I didn't remove it right away but have it throw an error now

**Review by juliasilge — APPROVED** [review]
This looks great 🙌 

If we are keeping all three kPCA steps, that puts us at "copying three times" and I say let's put the info in a roxygen template. The template is technically incorrect right now for `step_kpca()` (no `keep_original_cols = FALSE`) but I will work on that immediately after this is merged in.

I'll let you merge @hfrick after you look at my work here.

**Comment by hfrick** [comment]
Oh, good catch! Thanks for adding that. :raised_hands: 

I've added a single "the" to the sentence and then it cascaded across the standard text width so it looks like a lot more changes than that :eyes: 

I think this is ready to be merged. I'll do that once the checks are finished

**Pull request merged**
